### PR TITLE
inspection is looking for Sheets.Item and Sheets._Default member calls

### PR DESCRIPTION
**Pull request by retailcoder**

### CODE REVIEW

File: RubberduckTests/Inspections/SheetAccessedUsingStringInspectionTests.cs [hunk 1]
```diff
@@ -11,9 +11,11 @@
 
 namespace RubberduckTests.Inspections
 {
+    [Ignore("TODO: figure out how to mock retrieving the document module supertypes from the usercode ITypeLib; none of these tests are conclusive until then...")]
```
Comment (MDoerner) [inline RubberduckTests/Inspections/SheetAccessedUsingStringInspectionTests.cs:14]: Why not extract the current logic in the resolve runner base into an `ISuperTypeNamesForDocumentsProvider` with a single method taking a project id and a collection of document declarations and returning a dictionary document module - > super type names? That could then be injected and mocked in the tests.

File: Rubberduck.CodeAnalysis/Inspections/Concrete/SheetAccessedUsingStringInspection.cs [hunk 2]
```diff
@@ -78,20 +72,59 @@ protected override IEnumerable<Declaration> ObjectionableDeclarations(Declaratio
...
         protected override (bool isResult, string properties) IsResultReferenceWithAdditionalProperties(IdentifierReference reference, DeclarationFinder finder)
         {
+            var isHostWorkbookQualifier = false;
+            var hostWorkbookDeclaration = GetHostWorkbookDeclaration(finder);
+            var appObjectDeclaration = GetHostApplicationDeclaration(finder);
+
+            var context = reference.Context.Parent;
+            if (context is VBAParser.MemberAccessExprContext memberAccess)
+            {
+                isHostWorkbookQualifier = memberAccess.lExpression().GetText().Equals(hostWorkbookDeclaration.IdentifierName, System.StringComparison.InvariantCultureIgnoreCase);
```
[the remark this answers is not in the document]
Comment (retailcoder) [reply Rubberduck.CodeAnalysis/Inspections/Concrete/SheetAccessedUsingStringInspection.cs:113]: Good points; note that this particular inspection should only run when hosted in Excel though.

File: Rubberduck.Parsing/VBA/DeclarationResolving/DeclarationResolveRunnerBase.cs [hunk 3]
```diff
@@ -144,6 +144,9 @@ protected void ResolveDeclarations(QualifiedModuleName module, IParseTree tree,
                     .GroupBy(a => a.AnnotatedLine.Value)
                     .ToDictionary();
                 var attributes = _state.GetModuleAttributes(module);
+                if (module.ComponentType == ComponentType.Document)
```
[the remark this answers is not in the document]
Comment (retailcoder) [reply Rubberduck.Parsing/VBA/DeclarationResolving/DeclarationResolveRunnerBase.cs:147]: Argh, I must have accidentally included diffs of the hacking attempts. This file/diff shouldn't be in this PR.

File: Rubberduck.Parsing/VBA/DeclarationResolving/SynchronousDeclarationResolveRunner.cs [hunk 4]
```diff
@@ -35,6 +35,20 @@ protected override void ResolveDeclarations(IReadOnlyCollection<QualifiedModuleN
                 foreach(var module in modules)
                 {
                     ResolveDeclarations(module, _state.ParseTrees.Find(s => s.Key == module).Value, projects, token);
+                    var declaration = _state.DeclarationFinder.ModuleDeclaration(module);
```
[the remark this answers is not in the document]
Comment (retailcoder) [reply Rubberduck.Parsing/VBA/DeclarationResolving/SynchronousDeclarationResolveRunner.cs:38]: Seemed much easier than doing the actual right thing, but it doesn't work and shouldn't have been in this diff.

File: RubberduckTests/Inspections/SheetAccessedUsingStringInspectionTests.cs (hunk 1 above)
Comment (retailcoder) [reply RubberduckTests/Inspections/SheetAccessedUsingStringInspectionTests.cs:14]: On it.

**Pull request merged**